src/poetry/utils/helpers.py

```diff
@@ -133,6 +138,11 @@ def download_file(
                             last_percent = percent
                             update_context(f"Downloading {url} {percent:3}%")
 
+def download_file_with_curl(
+    url: str,
+    dest: str,
+) -> None:
+    subprocess.run(['curl', url, '--silent', '--output', dest])
```

Comment:
now that you found that requests/urllib3 doesn't have this problem, is it possible to use `requests` and convince it to not use the cache? the main reason is that this breaks packages that are in private repositories (i.e. if `session` is present, it will be ignored).

also, it's better if we don't rely on an external binary for this ^^;; also, `subprocess.run` fails silently, you'd need to add `check=True` for it to throw an exception if the command failed.

Reply:
:+1: we can do that.